sources/models/src/de.rs

```diff
@@ -1,8 +1,130 @@
-use crate::RegistryMirror;
+use crate::{KubernetesLabelKey, KubernetesTaintValue, RegistryMirror};
 use serde::de::value::SeqAccessDeserializer;
-use serde::de::{MapAccess, SeqAccess, Visitor};
+use serde::de::{Error, MapAccess, SeqAccess, Visitor};
 use serde::{Deserialize, Deserializer};
+use std::collections::HashMap;
+use std::convert::TryFrom;
 use std::fmt::Formatter;
+use toml::Value;
+
+// Our standard representation of node-taints is a `HashMap` of label keys to a list of taint values and effects;
+// for backward compatibility, we also allow a `HashMap` of label keys to a singular taint value/effect.
+pub(crate) fn deserialize_node_taints<'de, D>(
+    deserializer: D,
+) -> Result<Option<HashMap<KubernetesLabelKey, Vec<KubernetesTaintValue>>>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    struct NodeTaints;
+
+    impl<'de> Visitor<'de> for NodeTaints {
+        type Value = Option<HashMap<KubernetesLabelKey, Vec<KubernetesTaintValue>>>;
+        fn expecting(&self, formatter: &mut Formatter) -> std::fmt::Result {
+            formatter.write_str("TOML table")
+        }
+
+        fn visit_map<M>(self, mut map: M) -> Result<Self::Value, M::Error>
+        where
+            M: MapAccess<'de>,
+        {
+            let mut node_taints: HashMap<KubernetesLabelKey, Vec<KubernetesTaintValue>> =
+                HashMap::new();
+            while let Some((k, v)) = map.next_entry()? {
+                match v {
+                    // If we encounter a singular mapped value, convert it into a list of one
+                    Value::String(taint_val) => {
+                        node_taints.insert(
+                            k,
+                            vec![KubernetesTaintValue::try_from(taint_val)
+                                .map_err(M::Error::custom)?],
+                        );
+                    }
+                    // If we encounter a list of values, just insert it as is
+                    Value::Array(taint_values) => {
+                        let taint_values = taint_values
+                            .iter()
+                            .map(|v| v.to_owned().try_into().map_err(M::Error::custom))
+                            .collect::<Result<Vec<KubernetesTaintValue>, _>>()?;
+                        node_taints.insert(k, taint_values);
+                    }
```

Comment:
Does this catch the case of an empty list as input?

```
[settings.kubernetes.node-taints]
"special" = []
```

Reply:
Oh good catch.  Hmm, I guess we should error if that's the case since that would be an invalid node-taint